File: Lyapunov.py
import numpy as np
import math
import random
x=[]
y=[]
lim = 10000
init = random.uniform(0.001,0.999)
sum_sigma=0
for r in np.arange(1,4,0.001):
    for i in range(0,lim):
        init = r*(1-init)*init  #現在の値を次の世代に更新する
        diff = r*(1 - 2*init) #微分
        sum_sigma += np.log(abs(diff)) #微分した値を対数にして足し算する
    sum_sigma = sum_sigma / lim  #リアプノフ指数は過去の値に影響受けるため
    x.append(r)
    y.append(sum_sigma)

File: Lyapunov.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>
int main(void) {
  double sum_sigma=0;
  int lim = 10000;
  srand(time(NULL));
  FILE *file;
	file = fopen("c_lyapunov.txt","w");
  double init = (double)rand()/RAND_MAX;
  for(float r=1.0; r<4.0; r+=0.001){
    for(int j=0; j<lim; j++){
      init = r*(1-init)*init;
      double diff = r*(1 - 2*init);
      sum_sigma += log(fabs(diff));
    }
    sum_sigma = sum_sigma / lim;
    fprintf(file, "%f",r);
    fprintf(file,",");
    fprintf(file, "%f",sum_sigma);
    fprintf(file,"\n");
  }
  fclose(file);
  printf("終了しました");
  return 0;
}
